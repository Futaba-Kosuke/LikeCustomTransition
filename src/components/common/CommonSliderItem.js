/**
 * キャンペーンリスト表示用UIのベース部分を表示するコンポーネント
 */
import React from 'react';

//ReactNativeを使用したコンポーネントの呼び出し
import {
  StyleSheet,
  TouchableWithoutFeedback,
  View,
  Image,
  Dimensions,
  Text,
} from 'react-native';

//デバイスのサイズ取得
const {
  width: DEVICE_WIDTH,
} = Dimensions.get('window');

//ギャラリーの幅と高さの設定
const sliderWidth = DEVICE_WIDTH;
const sliderHeight = DEVICE_WIDTH / 2;

//一覧シンプルリスト表示用のベースコンポーネントの内容を定義する
const CommonSliderItem = ({ slider, onPress }) => {

  //取得した引数（オブジェクト：{ slider }）を分割する
  const { id, title, image_url } = slider;

  //表示する要素を返す
  return (
    <View style={styles.slideThumbnailContainerStyle}>
      <Image style={styles.slideThumbnailStyle} source={{ uri: image_url }} />
      <TouchableWithoutFeedback onPress={onPress}>
        <View style={styles.overlayStyle}>
          <Text style={styles.overlayTitleStyle}>{title}</Text>
          <Text style={styles.overlayCategoryStyle}>▶︎ 記事詳細へ</Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  );
};

//このコンポーネントのStyle定義
const styles = {
  slideThumbnailContainerStyle: {
    height: sliderHeight,
    backgroundColor: '#F5FCFF',
    position: 'relative',
  },
  slideThumbnailStyle: {
    width: sliderWidth,
    height: sliderHeight,
    backgroundColor: '#eee',
  },
  overlayStyle: {
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    paddingLeft: 15,
    paddingRight: 15,
    paddingTop: 24,
    paddingBottom: 20,
    backgroundColor: 'rgba(0, 0, 0, 0.6)',
  },
  overlayTitleStyle: {
    fontSize: 14,
    fontWeight: 'bold',
    color: '#fff',
  },
  overlayCategoryStyle: {
    marginTop: 5,
    fontSize: 12,
    color: '#fff',
  },
};

//このコンポーネントをインポート可能にする
export default CommonSliderItem;
