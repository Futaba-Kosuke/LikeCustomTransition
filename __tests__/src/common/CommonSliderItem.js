import 'react-native';
import React from 'react';
import CommonSliderItem from '../../../src/components/common/CommonSliderItem';
import { getSliderList } from '../../../src/stub/SampleDataStub';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('共通コンポーネント(コラムのスライダー表示用部分)のスナップショットテスト', () => {
  const slider = getSliderList();
  const tree = renderer.create(
    <CommonSliderItem slider={slider[0]} onPress={ () => console.log("テスト") } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
