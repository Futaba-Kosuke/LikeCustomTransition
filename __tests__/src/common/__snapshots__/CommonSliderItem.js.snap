// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`共通コンポーネント(コラムのスライダー表示用部分)のスナップショットテスト 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "height": 375,
      "position": "relative",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://s3-ap-northeast-1.amazonaws.com/otsuka-shop/images/column/column1.jpg",
      }
    }
    style={
      Object {
        "backgroundColor": "#eee",
        "height": 375,
        "width": 750,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "bottom": 0,
        "left": 0,
        "paddingBottom": 20,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 24,
        "position": "absolute",
        "right": 0,
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      コラムタイトル1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 12,
          "marginTop": 5,
        }
      }
    >
      ▶︎ 記事詳細へ
    </Text>
  </View>
</View>
`;
